### ISSUE TITLE
Support ref.focus() alias

### ISSUE THREAD

**Issue by danilofuchs** [Hacker0x01/react-datepicker#4834]
**Is your feature request related to a problem? Please describe.**
When using `react-hook-form`, it's ability to focus on the input automatically on error uses `ref.focus()`.

However, react-datepicker uses the nomenclature `ref.setFocus()`

**Describe the solution you'd like**
An alias `ref.focus()` to allow better compatibility with the ecosystem

**Describe alternatives you've considered**
- Intercepting the ref myself, calling `ref.setFocus()`:
```typescript
    const innerRef = useRef<ReactDatePicker>(null);
    
    useImperativeHandle(ref, () => ({
      focus: () => innerRef.current?.setFocus(),
    }));
    
    return (
      <ReactDatePicker
        ref={innerRef}
      />
    )
```
- Deprecating `setFocus()`, `setBlur()`, etc.
- Use `customInputRef`, but it only allows legacy string refs

**Additional context**
https://react-hook-form.com/docs/useform/setfocus
![image](https://github.com/Hacker0x01/react-datepicker/assets/17657014/c1288747-a5ec-40cf-82d9-d74df1dc5b35)

**Comment by KubaZachacz** [comment]
Yes, would be great, I've lost a lot of time until I came up with useImperativeHandle solution
